framework/src/consumption/lib/redshift-serverless/redshift-serverless-workgroup.ts

```diff
@@ -0,0 +1,268 @@
+ *   ]
+ * })
+ *
+ * const workgroup = new dsf.consumption.RedshiftServerlessWorkgroup(this, "RedshiftWorkgroup", {
+ *    vpc,
+ *    subnets: {
+ *      subnetGroupName: "private-dwh"
+ *    },
+ *    workgroupName: "redshift-workgroup"
+ * })
+ */
+export class RedshiftServerlessWorkgroup extends TrackedConstruct implements IConnectable {
+  /**
+   * The created workgroup
+   */
+  readonly cfnResource: CfnWorkgroup;
+
+  /**
+   * The name of the created workgroup
+   */
+  readonly workgroupName: string;
+
+  /**
+   * Connections used by Workgroup security group. Used this to enable access from clients connecting to the workgroup
+   */
+  readonly connections: Connections;
+
+  /**
+   * The associated namespace
+   */
+  readonly namespace: RedshiftServerlessNamespace;
+
+  /**
+   * The name of the associated namespace
+   */
+  readonly namespaceName: string;
+
+  /**
+   * The Glue connection associated with the workgroup. This can be used by Glue ETL Jobs to read/write data from/to Redshift workgroup
+   */
+  readonly glueConnection: CfnConnection;
+
+  private readonly primarySecurityGroup: SecurityGroup;
+  private readonly removalPolicy: RemovalPolicy;
+  private redshiftData?: RedshiftData;
+  private readonly vpc?: Vpc;
+  private readonly vpcSubnets?: SubnetSelection;
+
+  constructor(scope: Construct, id: string, props: RedshiftServerlessWorkgroupProps) {
+    const trackedConstructProps: TrackedConstructProps = {
+      trackingTag: RedshiftServerlessWorkgroup.name,
+    };
+
+    super(scope, id, trackedConstructProps);
+    this.removalPolicy = Context.revertRemovalPolicy(scope, props.removalPolicy);
+    const initNamespaceDetails = this.initNamespace(props);
+    this.namespace = initNamespaceDetails.namespace;
+    this.namespaceName = initNamespaceDetails.namespaceName;
+
+    const initSecurityGroupDetails = this.initSecurityGroup(props);
+    this.primarySecurityGroup = initSecurityGroupDetails.primarySecurityGroup;
+    this.connections = initSecurityGroupDetails.primaryConnections;
+
+    const selectedSubnets = props.vpc.selectSubnets(props.subnets);
+    const subnetIds = selectedSubnets.subnetIds;
+    this.workgroupName = `${props.workgroupName}-${Names.uniqueResourceName(this, { maxLength: 63 - (props.workgroupName.length + 1) }).toLowerCase()}`;
+
+
+    this.cfnResource = new CfnWorkgroup(
+      this,
+      'RSServerlessWorkgroup',
+      {
+        workgroupName: this.workgroupName,
+        baseCapacity: props.baseCapacity,
+        enhancedVpcRouting: true,
+        namespaceName: this.namespaceName,
+        port: props.port,
+        publiclyAccessible: false,
+        subnetIds,
+        securityGroupIds: initSecurityGroupDetails.securityGroupIds,
+      },
+    );
+
+    this.cfnResource.applyRemovalPolicy(this.removalPolicy);
+    this.cfnResource.node.addDependency(this.namespace.cfnResource);
+    this.glueConnection = this.createConnection(selectedSubnets, props);
+    this.vpc = props.vpc;
```

Comment:
We should have an option to create a VPC on behalf of the user if he does not provide one.

Reply:
Done. Updated the following (in addition to the construct code):

- Tests
- Example